rpc/block_svc_routines.c

```diff
@@ -2774,6 +2946,265 @@ block_modify_cli_1_svc_st(blockModifyCli *blk, struct svc_req *rqstp)
+  if (!glfs) {
+    LOG("mgmt", GB_LOG_ERROR,
+        "glusterBlockVolumeInit(%s) for block %s failed [%s]",
+        blk->volume, blk->block_name, strerror(errno));
+    goto initfail;
+  }
+
+  lkfd = glusterBlockCreateMetaLockFile(glfs, blk->volume, &errCode, &errMsg);
+  if (!lkfd) {
+    LOG("mgmt", GB_LOG_ERROR, "%s %s for block %s",
+        FAILED_CREATING_META, blk->volume, blk->block_name);
+    goto nolock;
+  }
+
+  GB_METALOCK_OR_GOTO(lkfd, blk->volume, ret, errMsg, nolock);
+
+  if (glfs_access(glfs, blk->block_name, F_OK)) {
+    errCode = errno;
+    if (errCode == ENOENT) {
+      GB_ASPRINTF(&errMsg, "block %s/%s doesn't exist",
+                  blk->volume, blk->block_name);
+      LOG("mgmt", GB_LOG_ERROR,
+          "block with name %s doesn't exist in the volume %s",
+          blk->block_name, blk->volume);
+    } else {
+      GB_ASPRINTF(&errMsg, "block %s/%s is not accessible (%s)",
+                  blk->volume, blk->block_name, strerror(errCode));
+      LOG("mgmt", GB_LOG_ERROR, "block %s/%s is not accessible (%s)",
+          blk->volume, blk->block_name, strerror(errCode));
+    }
+    goto out;
+  }
+
+  ret = blockGetMetaInfo(glfs, blk->block_name, info, NULL);
+  if (ret) {
+    goto out;
+  }
+
+  if (info->size > blk->size && !blk->force) {
+    cSize = glusterBlockFormatSize("mgmt", info->size);
+    rSize = glusterBlockFormatSize("mgmt", blk->size);
+    GB_ASPRINTF(&errMsg, "Shrink size ?\nuse 'force' option [current size %s, request size %s]",
+                cSize, rSize);
+    GB_FREE(cSize);
+    GB_FREE(rSize);
+    errCode = -1;
+    goto out;
+  }
+
+  list = glusterBlockGetListFromInfo(info);
+  if (!list) {
+    errCode = ENOMEM;
+    goto out;
+  }
+
+  errCode = glusterBlockCheckCapabilities((void *)blk, MODIFY_SIZE_SRV, list, &errMsg);
+  if (errCode) {
+    LOG("mgmt", GB_LOG_ERROR,
+        "glusterBlockCheckCapabilities() for block %s on volume %s failed",
+        blk->block_name, blk->volume);
+    goto out;
+  }
+
+  GB_STRCPYSTATIC(mobj.block_name, blk->block_name);
+  GB_STRCPYSTATIC(mobj.volume, blk->volume);
+  GB_STRCPYSTATIC(mobj.gbid, info->gbid);
+  mobj.size = blk->size;
+
+  ret = glusterBlockResizeEntry(glfs, &mobj, &errCode, &errMsg);
+  if (ret) {
+    LOG("mgmt", GB_LOG_ERROR, "%s block: %s volume: %s file: %s size: %zu",
+        FAILED_MODIFY_SIZE, mobj.block_name, mobj.volume, mobj.gbid, mobj.size);
+    ret = errCode;
+    goto out;
+  }
+
+  asyncret = glusterBlockModifySizeRemoteAsync(info, glfs, &mobj, &savereply);
+  if (asyncret) {   /* asyncret decides result is success/fail */
+    errCode = asyncret;
+    LOG("mgmt", GB_LOG_WARNING,
+        "glusterBlockModifySizeRemoteAsync(size=%zu): return %d %s for block %s on volume %s",
+        blk->size, asyncret, FAILED_REMOTE_AYNC_MODIFY, blk->block_name, info->volume);
+    goto out;
+  } else {
+    GB_METAUPDATE_OR_GOTO(lock, glfs, mobj.block_name, mobj.volume,
```

Comment:
Not sure, if we want to do this in else block only.

Reply:
no problem I think